lib/db_utils.pm

```diff
@@ -61,18 +66,25 @@ Example of data:
 sub influxdb_push_data {
     my ($url, $db, $org, $token, $data, %args) = @_;
     $args{quiet} //= 1;
+    $args{proceed_on_failure} //= 0;
     $data = build_influx_query($data);
     my $cmd = sprintf("curl -iLk -X POST '$url/api/v2/write?org=$org&bucket=$db' --header 'Authorization: Token $token' --write-out 'RETURN_CODE:%%{response_code}' --data-binary '%s'", $data);
 
     # Hide the token in the info box
     my $out = $cmd;
     $out =~ s/$token/<redacted>/;
-    record_info('curl', $out);
+    record_info('curl POST', $out);
 
     my $output = script_output($cmd, quiet => $args{quiet});
     my ($return_code) = $output =~ /RETURN_CODE:(\d+)/;
-    die("Fail to push data into Influx DB:\n$output") unless ($return_code >= 200 && $return_code < 300);
-}
+    unless ($return_code >= 200 && $return_code < 300) {
+        my $msg = "Failed pushing data into Influx DB:\n$output\n";
+        record_info("FAIL push db", $msg, result => 'fail' ) if ($args{proceed_on_failure});
+        die($msg) unless ($args{proceed_on_failure});
+        return;
+    } # endif
+    return 1;
+} # endsub
```

Comment:
Indeed it is only in this sub, but I added this comment as programming style in each end block, that I usually preferred to see as help in the code inspection, specific for languages using parenthesys.

Reply:
if we start putting such comments everywhere in the code , it will too much noise . Also any modern IDE is capable to solve this problem without any comments ( it is called "parenthesis highlight" :) )

Reply:
ok